cmd/fleet/serve.go

```diff
@@ -499,89 +499,120 @@ the way that the Fleet server works.
 				mdmPushService = nanomdm_pushsvc.New(mdmStorage, mdmStorage, pushProviderFactory, nanoMDMLogger)
 			}
 
-			// validate Apple APNs/SCEP config
+			checkMDMAssets := func(names []fleet.MDMAssetName) (bool, error) {
+				_, err = ds.GetAllMDMConfigAssetsByName(context.Background(), names)
+				if err != nil {
+					if fleet.IsNotFound(err) || errors.Is(err, mysql.ErrPartialResult) {
+						return false, nil
+					}
+					return false, err
+				}
+				return true, nil
+			}
+
+			// reconcile Apple Business Manager configuration environment variables with the database
 			if config.MDM.IsAppleAPNsSet() || config.MDM.IsAppleSCEPSet() {
 				if !config.MDM.IsAppleAPNsSet() {
 					initFatal(errors.New("Apple APNs MDM configuration must be provided when Apple SCEP is provided"), "validate Apple MDM")
 				} else if !config.MDM.IsAppleSCEPSet() {
 					initFatal(errors.New("Apple SCEP MDM configuration must be provided when Apple APNs is provided"), "validate Apple MDM")
 				}
 
-				if len(config.Server.PrivateKey) == 0 {
-					initFatal(errors.New("inserting APNs and SCEP assets"), "missing required private key. Learn how to configure the private key here: https://fleetdm.com/learn-more-about/fleet-server-private-key")
-				}
```

Comment:
I think the reason for this check is that reading the MDM assets (the call to `ds.GetAllMDMConfigAssetsByName` in `checkMDMAssets`) [requires that private key](https://github.com/fleetdm/fleet/blob/9f7bbfe065bac13bccaf44c28acf73e76b8ead7d/server/datastore/mysql/apple_mdm.go#L4912-L4919), as the assets are encrypted with it in the database.

So if it is not set, it will likely still fail to start but with a more confusing error message (failing to decrypt probably).